### PR TITLE
Update cert inheritance tests to match new xapi behavior

### PR DESCRIPTION
**Pull request by benjamreis**
For XCP-ng > 8.2.1 there is a new behavior regarding SB certs management
See: https://github.com/xapi-project/xen-api/pull/4659

Keep previous behavior for XCP-ng <= 8.2.1

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

### CODE REVIEW

**Review by gduperrey — APPROVED** [review]

File: tests/uefistored/test_secure_boot.py
Comment (stormi) [inline tests/uefistored/test_secure_boot.py:406]: Testing for 8.2.1 means that whenever we release 8.2.2 it will be expected that it has the new behaviour. This is not guaranteed. I'd test for > 8.2.99 or >= 8.3

File: tests/uefistored/test_secure_boot.py
Comment (stormi) [inline tests/uefistored/test_secure_boot.py:423]: I think we probably don't need to start a VM at all anymore in TestPoolToDiskCertInheritance (except maybe in one dedicated test where we'd test that starting the VM without a toolstack restart does *not* update the certs on disk?)

File: tests/uefistored/test_secure_boot.py
Comment (benjamreis) [reply tests/uefistored/test_secure_boot.py:406]: True

File: tests/uefistored/test_secure_boot.py
Comment (benjamreis) [reply tests/uefistored/test_secure_boot.py:423]: We need the start to get the `resident_on` later in the code.

File: tests/uefistored/test_secure_boot.py
Comment (benjamreis) [reply tests/uefistored/test_secure_boot.py:389]: i'll use the packaging.version lib.

File: tests/uefistored/test_secure_boot.py
Comment (stormi) [reply tests/uefistored/test_secure_boot.py:423]: Unless you use the `host` fixture

**Review by stormi — COMMENTED** [review]
It would be good to add a docstring to TestPoolToDiskCertInheritance explaining why there are two different behaviours in the tests, what's the old behaviour and what's the new one.

**Review by stormi — APPROVED** [review]
Approved, but let's wait for the XAPI PR to be merged before we merge this.

File: tests/uefistored/test_cert_inheritance.py
Comment (stormi) [inline tests/uefistored/test_cert_inheritance.py:9]: There should be a "requirements" comment here (which should document the requirement of at least 2 hosts in the pool for TestPoolToDiskCertInheritance).

File: lib/pool.py
Comment (stormi) [inline lib/pool.py:144]: I don't think this is useful, as you only copy the reference.
```
>>> a = [1,2]
>>> b = a
>>> b.append(3)
>>> a
[1, 2, 3]
```

However, AFAICT, in the next line self.hosts will only be written after `[h for h in hosts if h.uuid != host.uuid]` so there's no risk of modifying the list while it's being iterated.

File: lib/pool.py
Comment (stormi) [inline lib/pool.py:143]: Not entirely sure but I would write "from".

```suggestion
        wait_for_not(lambda: host.uuid in self.hosts_uuids(), "Wait for host {host} to be ejected from pool {master}.")
```

File: tests/uefistored/test_cert_inheritance.py
Comment (stormi) [inline tests/uefistored/test_cert_inheritance.py:343]:
```suggestion
            pytest.skip(f"This test requires a second XCP-ng pool with version >= {version_str} host")
```

File: tests/uefistored/test_cert_inheritance.py
Comment (stormi) [inline tests/uefistored/test_cert_inheritance.py:356]:
```suggestion
        # Warning: triggers a reboot of ejected host.
```

File: lib/pool.py
Comment (benjamreis) [reply lib/pool.py:144]: i'll try

**Review by stormi — APPROVED** [review]
Approved... Now we still depend on the XAPI PR.

**Pull request merged**
